src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2394,7 +2394,7 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
     if runtime_version is not None:
         if runtime is None:
             raise CLIError('Must specify --runtime to use --runtime-version')
-        allowed_versions = RUNTIME_TO_IMAGE_FUNCTIONAPP[functions_version][runtime].keys()
+        allowed_versions = RUNTIME_TO_SUPPORTED_VERSIONS_FUNCTIONAPP[functions_version][runtime]
         if runtime_version not in allowed_versions:
```

Comment:
Do you do any validate for --runtime-version for `Windows` function apps?

Reply:
Apologies if I misunderstood the question -- the `if` block before this one will check that the `runtime` they've selected is valid for the OS, and this block will check that the `runtime_version` is valid for the `runtime` regardless of the OS. As far as I understand (at least to this point) supported `runtime_version`'s don't change across OS's. They're only different across `functions_version`'s. So as long as the `runtime` is supported for the OS, the list of supported `runtime_version`'s should be the same.

Reply:
Thanks for your reply. There are some runtime versions that are only supported on FunctionsVersion 2, and not in 3 and vice versa. For the cmdlets, I will validate against a static json file that @pragnagopa is working on. It would probably make sense that all clients (`az cli`, `Portal`, and `Az.Functions cmdlets`) do the same. However, that does not apply in this PR. This is more like a future work item.

Reply:
Right so this will do the validation to make sure the `runtime_version` is valid for the `functions_version`. And yeah using the json files pragna is working on is definitely planned for the cli